### PR TITLE
[JENKINS-68303] Animate button when job already in queue

### PR DESCRIPTION
**Pull request by alecharp**
In #5851, the `icon-clock-anime` was removed in favor of the `play-outline` icon. However, it removed the animation we had when the job was already in the build queue.

Bildschirmfoto 2022-05-03 um 19 46 37

With this PR, we are getting back the animation on the schedule button when the job is present on the build queue:

https://user-images.githubusercontent.com/13383509/166511500-7aebe7e9-6b21-4f26-bc80-af3e1e935632.mp4

See [JENKINS-68303](https://issues.jenkins.io/browse/JENKINS-68303).

### Proposed changelog entries

* Fix indistinguishable build scheduling icon when the job is already in-queue.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@janfaracik @jenkinsci/sig-ux 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

### CODE REVIEW

File: war/src/main/less/modules/icons.less
Comment (NotMyFault) [inline war/src/main/less/modules/icons.less:129]:
```suggestion
@keyframes pulse-animation {
  0%,
  100% {
    opacity: 1;
  }
  50% {
    opacity: 0;
  }
}

.pulse-animation {
  svg {
    animation: pulse-animation 2s ease infinite;
  }
}
```

Untested (like your version 👀 ) but it turns your proposal into an agnostic class that can be reused in other occasions, if needed, and makes the animation more calm. This change simulates a pulse, instead of having the icon pop up instant after the fadeout.

Needs to be adjusted in the jelly file too, if we want to use it.

**Comment by uhafner** [comment]
Can you please add a before and after screen recording?

**Comment by NotMyFault** [comment]
> Can you please add a before and after screen recording?

I've added one and one for my suggestion too.

File: war/src/main/less/modules/icons.less
Comment (alecharp) [reply war/src/main/less/modules/icons.less:129]: This is way better. Thank you @NotMyFault !

**Comment by alecharp** [comment]
Thank you @NotMyFault for the screenshot. Using the `pulse-animation`, the return of the icon is even smoother than on your video clip.

**Review by NotMyFault — APPROVED** [review]
Nice one

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Review by janfaracik — APPROVED** [review]

**Comment by judovana** [comment]
Thanx a lot guys!

**Comment by uhafner** [comment]
Seems that the *before* image works, but the *after* video is broken somehow. I think I need to install the PR locally...

Did someone check if the whole table is blinking if there are a lot of jobs in the queue? I think we should use a blinking style only in very rare occasions (or make that a column that can be added by a user that wants such a feature). If possible we should focus on the real requirement: make it visible that a job is in the queue.

**Comment by judovana** [comment]
That is interesting idea. One argument against is to conserve space. Why to have one additional column (see https://issues.jenkins.io/browse/JENKINS-68304?filter=-2 ) when the schedule button can have multiple states.  As this button serves for scheduling, it sounds jut right to show state of scheduling. 
Instead of blinking, some other animation can be used - eg the for ever drawn arrow, as is is for ever drawn circle of "is building" or similarly.

The whole blinking table is good feature. It may happen that some set of jobs is stuck in queue, and it is by luck discovered in some view. Where query for mass schedule is quite straightforward, the queries for jobs in queue or  being built are not so straightforward.

" focus on the real requirement: make it visible that a job is in the queue." is well worded. Thanx!

**Comment by alecharp** [comment]
I understand that we might want to rethink this all "highlight jobs in the build queue" but here I just wanted to bring back what we had in <`2.321`. Having another column could unnecessarily increase the density of the table.

> Did someone check if the whole table is blinking if there are a lot of jobs in the queue?

I tested with 3 jobs and yes, it does.

**Comment by uhafner** [comment]
I still don't see any difference locally? Where and when exactly should the change show up? 


https://user-images.githubusercontent.com/503338/166675652-028e995f-0c84-47c4-a0ca-f485e050b1c6.mov


When a build is in the queue then the build status icon already has an animation. Why should the run button have another one? 

> Having another column could unnecessarily increase the density of the table.

Well this button should be a different column that can be added by a user if they really want a blinking and flashing old-school UI back (then this user can remove the non-blinking button). But I think the default should be a clean and concise UI that uses only a minimum set of distortions.

**Comment by alecharp** [comment]
@uhafner if you install the schedule build plugin, you can make the job stay in quiet mode for a long period. This way it's easier to see the pulse animation.

**Comment by judovana** [comment]
I would highly recommend to first fix the regression, and then revisit the new colum and/or possible states of the green arrow button.

**Comment by judovana** [comment]
> @uhafner if you install the schedule build plugin, you can make the job stay in quiet mode for a long period. This way it's easier to see the pulse animation.

Or you can run on non-existing slave and so keep itin queue for ever.

**Comment by uhafner** [comment]
I won't consider the UI improvement we have in recent Jenkins versions a regression, but anyway.  

> Or you can run on non-existing agent and so keep itin queue for ever.

Hmm, my agent is offline. And still no animation visible. Maybe someone can post a new screencast?

**Comment by judovana** [comment]
> I won't consider the UI improvements we have in recent Jenkins versions a regression, but anyway.

Don't take me bad. The new UI is awesome peace of work.

**Comment by uhafner** [comment]
> When a build is in the queue then the build status icon already has an animation. Why should the run button have another one?

Can someone ask this question? Is the state you are describing in this PR a different one? I.e., the status is not animating but the build button does? Or are we adding a second animation?

**Comment by alecharp** [comment]
> I won't consider the UI improvements we have in recent Jenkins versions a regression, but anyway.

Definitively not that. It's just the animation was in place before the improvement of the table. In this improvement, the `gif` files were removed (which is ok) but the new icon used was never animated where we had a specific `gif` for the "job is in the build queue" situation.

**Comment by judovana** [comment]
> > When a build is in the queue then the build status icon already has an animation. Why should the run button have another one?
> 
> Can someone answer this question? Is the state you are describing in this PR a different one? I.e., the status is not animating but the build button does? Or are we adding a second animation?

Whether the build is in queue or is running, are two completely different states, which should be visible by two different icons xor animations.

It is natural (read legayi habit)  to see scheduled state visible on schedule button, and the running job on last result  icon.In addition  that  the build can be both running and scheduled, so you need two icons to show four states (not running and not scheduled, scheduled not running, running not scheduled , running and scheduled).

Still some possibility to show 4 states in the last result build is an option. stable, animated, animated an blinking,  differnelty animated...  But thenyou can discuss that the schedule button in defuatl view is  moreover useless.

My use case relays  with 1000+ memebrs in queue, where are tasks ffrm few seconds to *week* of runtime. The posibility todistinguish  is a must.

**Comment by judovana** [comment]
Just for record, I built the jenkisn with this patch and it works as expected - the green arrow is blinking if the build is scheduled. Works fine in various views, and had not spot any side effect.

**Comment by alecharp** [comment]
@uhafner to be able to track your point, which is totally valid (blinking is not the best), I created https://issues.jenkins.io/browse/JENKINS-68413 and I think we could discuss this in the next SIG meeting. Is that ok for you?

**Review by basil — APPROVED** [review]
Thanks to everyone for the very relevant discussion. I am going to merge this simple regression fix as-is, but I concur that this functionality was never great in the first place and has room for improvement. Restoring the status quo buys us time to make further improvements without negatively impacting existing use cases.

**Pull request merged**
